File: scala/src/com/runtimeverification/ktest/kthings/kitem.scala
package com.runtimeverification.ktest.kthings

import com.runtimeverification.ktest.KThing
import com.runtimeverification.ktest.kthings.block.Block

object kitem {
  class KItem extends KThing
  case class DivLeftMissing(value : aexp.Aexp) extends KItem
  case class DivRightMissing(value : aexp.Aexp) extends KItem
  case class AddLeftMissing(value : aexp.Aexp) extends KItem
  case class AddRightMissing(value : aexp.Aexp) extends KItem
  case class LessOrEqualsLeftMissing(value : aexp.Aexp) extends KItem
  case class LessOrEqualsRightMissing(value : aexp.Aexp) extends KItem

  case class NotOperandMissing() extends KItem
  case class AssignmentRightMissing(id : Id) extends KItem
  case class IfConditionMissing(ithen : Block, ielse : Block) extends KItem
}

File: java.optimized/src/main/java/com/runtimeverification/ktest/nonterminals/KItem.java
package com.runtimeverification.ktest.nonterminals;

import com.runtimeverification.ktest.KThing;

public class KItem {
    public static KThing divisionLeftMissing(KThing right) {
        return new KThing(KThing.Type.KITEM_DIVISION_LEFT_MISSING, right);
    }
    public static KThing divisionRightMissing(KThing left) {
        return new KThing(KThing.Type.KITEM_DIVISION_RIGHT_MISSING, left);
    }

    public static KThing additionLeftMissing(KThing right) {
        return new KThing(KThing.Type.KITEM_ADDITION_LEFT_MISSING, right);
    }
    public static KThing additionRightMissing(KThing left) {
        return new KThing(KThing.Type.KITEM_ADDITION_RIGHT_MISSING, left);
    }

    public static KThing lessOrEqualsLeftMissing(KThing right) {
        return new KThing(KThing.Type.KITEM_LESS_OR_EQUALS_LEFT_MISSING, right);
    }
    public static KThing lessOrEqualsRightMissing(KThing left) {
        return new KThing(KThing.Type.KITEM_LESS_OR_EQUALS_RIGHT_MISSING, left);
    }

    public static KThing notMissingOperand() {
        return new KThing(KThing.Type.KITEM_NOT_MISSING_OPERAND);
    }

    public static KThing andLeftMissing(KThing right) {
        return new KThing(KThing.Type.KITEM_AND_LEFT_MISSING, right);
    }

    public static KThing assignmentMissingOperand(KThing left) {
        return new KThing(KThing.Type.KITEM_ASSIGNMENT_MISSING_OPERAND, left);
    }

    public static KThing ifMissingCondition(KThing ithen, KThing ielse) {
        return new KThing(KThing.Type.KITEM_IF_MISSING_CONDITION, ithen, ielse);
    }
}
